tests/virtual_machines_test.go

```diff
@@ -82,49 +82,24 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				return len(vmList.Items)
 
 			}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
-
-			// remove all the labels from the test namespaces
-			for _, namespace := range []string{TestNamespace, OtherTestNamespace} {
-				err = cleanNamespaceLabels(namespace)
-				Expect(err).ToNot(HaveOccurred(), "should be able to remove the namespace labels")
-			}
-		})
-
-		Context("When the client wants to create a vm on an opted-out namespace", func() {
-			It("should create a vm object without a MAC assigned when no label is set to namespace", func() {
-				//remove namespace opt-in labels
-				By("removing the namespace opt-in label")
-				err := cleanNamespaceLabels(TestNamespace)
-				Expect(err).ToNot(HaveOccurred(), "should be able to remove the namespace labels")
-
-				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
-					[]kubevirtv1.Network{newNetwork("br")})
-
-				err = testClient.VirtClient.Create(context.TODO(), vm)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).Should(Equal(""))
-			})
 		})
 
 		Context("When kubemacpool is disabled on a namespace", func() {
 			It("should create a VM object without a MAC assigned", func() {
-				//change namespace opt-in label to disable
-				By("updating the namespace opt-in label to disabled")
-				err := cleanNamespaceLabels(TestNamespace)
-				Expect(err).ToNot(HaveOccurred(), "should be able to remove the namespace labels")
-				err = addLabelsToNamespace(TestNamespace, map[string]string{vmNamespaceOptInLabel: "disabled"})
+				By("opting out the namespace")
+				err := addLabelsToNamespace(TestNamespace, map[string]string{vmNamespaceOptInLabel: "ignore"})
 				Expect(err).ToNot(HaveOccurred(), "should be able to add the namespace labels")
 
 				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
 					[]kubevirtv1.Network{newNetwork("br")})
 
 				err = testClient.VirtClient.Create(context.TODO(), vm)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).Should(Equal(""))
+				Expect(err).ToNot(HaveOccurred(), "Should succeed creating the vm")
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).Should(Equal(""), "should not allocated a mac to the opted-out vm")
 			})
 		})
 
-		Context("When the client creates a vm on an opted-in namespace", func() {
+		Context("When the client creates a vm on a non-opted-out namespace", func() {
```

Comment:
And this context would be the default. Maybe `when all the VMs are opted-in by default`

The structure of these tests is confusing now. The main context should be "when all VMs are opted-in by default (implicit)". Then you would test that it gets VM. You would nest another Context in it "and when the VM is opted-in".

Reply:
Done